presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -1491,4 +1531,32 @@ private static void checkIsDoubled(ZoneId zone, LocalDateTime dateTime)
     {
         verify(zone.getRules().getValidOffsets(dateTime).size() == 2, "Expected %s to be doubled in %s", dateTime, zone);
     }
+
+    public static DataType<Float> postgreSqlRealDataType()
+    {
+        return dataType("real", RealType.REAL,
+                value -> {
+                    if (Float.isFinite(value)) {
+                        return value.toString();
+                    }
+                    if (Float.isNaN(value)) {
+                        return "'NaN'";
+                    }
+                    return format("'%sInfinity'", value > 0 ? "+" : "-");
+                });
+    }
+
+    public static DataType<Double> postgreSqlDoubleDataType()
+    {
+        return dataType("double precision", DoubleType.DOUBLE,
+                value -> {
+                    if (Double.isFinite(value)) {
+                        return value.toString();
+                    }
+                    if (Double.isNaN(value)) {
+                        return "'NaN'";
+                    }
+                    return format("'%sInfinity'", value > 0 ? "+" : "-");
+                });
```

Comment:
You could reuse `io.prestosql.testing.datatype.DataType#doubleDataType`, if you avoid nan/infinity "constants":

```
if (Double.isNaN(value)) {
    return "0 / 0";
}
return format("%s1 / 0", value > 0 ? "+" : "-");
```


(up to you)

Reply:
Would work I guess. I like the constants though. More readable IMO.